components/session-based-conditional-auth-functions/src/main/java/org/wso2/carbon/identity/extensions/authenticator/conditional/auth/functions/function/GetSessionDataFunction.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.exception.SessionValidationException;
+import org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.model.Session;
+import org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.util.SessionValidationConstants;
+import org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.util.SessionValidationUtil;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Represents javascript function provided in conditional authentication to retrieve active session data for given user.
+ * The purpose is to perform dynamic authentication selection based on the active session count.
+ */
+public class GetSessionDataFunction implements GetDataFunction {
+
+    private static final Log log = LogFactory.getLog(GetSessionDataFunction.class);
+
+    @Override
+    public JSONObject getData(JsAuthenticationContext context, Map<String, String> map) throws AuthenticationFailedException {
+
+        JSONObject jsonObject = new JSONObject();
+        JSONArray jsonArray = new JSONArray();
+        AuthenticatedUser authenticatedUser = context.getWrapped().getLastAuthenticatedUser();
```

Comment:
Does context.getWrapped method always return a value?

Reply:
AFAIK It is the only possible way to get the authenticated user from the conditional authentication context. This was discussed during the code review and it was suggested to keep it this way until getting authenticated user from context properties is supported.

Reply:
Then Its fine.